tests/linearizability/linearizability_test.go

```diff
@@ -96,23 +97,31 @@ func testLinearizability(ctx context.Context, t *testing.T, config e2e.EtcdProce
 		t.Fatal(err)
 	}
 	defer clus.Close()
-	ctx, cancel := context.WithCancel(ctx)
+	trafficCtx, trafficCancel := context.WithCancel(ctx)
 	go func() {
-		defer cancel()
-		err := triggerFailpoints(ctx, t, clus, failpoint)
-		if err != nil {
-			t.Error(err)
-		}
+		defer trafficCancel()
+		triggerFailpoints(ctx, t, clus, failpoint)
+		// Wait second to collect traffic after triggering last failpoint.
+		time.Sleep(time.Second)
 	}()
-	operations := simulateTraffic(ctx, t, clus, traffic)
+	watchCtx, watchCancel := context.WithCancel(ctx)
+	var operations []porcupine.Operation
+	go func() {
+		defer watchCancel()
+		operations = simulateTraffic(trafficCtx, t, clus, traffic)
+		// Wait second to collect watch events after all traffic was sent.
+		time.Sleep(time.Second)
+	}()
+	events := watchClusterChanges(watchCtx, t, clus)
```

Comment:
You'd better add comment to explain the relationship between three kinds of goroutines and how they are related by the cancel functions,
- The triggering failpoint goroutine
- The simulating traffic goroutine
- The watching event goroutines.

Reply:
Comments should explain "why?" and not "what?".  Isn't it clear in:
```
go func() {
		defer trafficCancel()
		triggerFailpoints(ctx, t, clus, failpoint)
		// Wait second to collect traffic after triggering last failpoint.
		time.Sleep(time.Second)
	}()
```
The failpoints gorouting cancels traffic?

Reply:
```
Isn't it clear
```

It's clear to you because you are the author. It doesn't mean it's clear to other contributors.  If I understood it correctly, the The **triggering failpoint goroutine** stops the **simulating traffic goroutine** by executing `defer trafficCancel()`, and the **simulating traffic goroutine** stops the **watching event goroutines** by **defer watchCancel()**. 

Note that it isn't Intuitive at first look. I was even thinking it might be a copy-paste typo, because I was confused why you use `ctx`, but defer cancel the `trafficCancel`. 

It isn't the first time I raise similar comment. Please always keep in mind to add as much self-contained comment as possible.

Reply:
> It's clear to you because you are the author. It doesn't mean it's clear to other contributors.

This is a very vague generic statement. Code is clean or it's not. If concept of using context to propagate stop signal is not familiar or standard then let's rewrite it. It's not about throwing as many comments as possible as it doesn't solve the problem with code.

Moved the code to separate function added comment explaining idea to use context to propagate cancellation.